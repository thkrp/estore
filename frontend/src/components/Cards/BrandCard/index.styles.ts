import styled from 'styled-components';
import Link from '../../Links/Link';
import { theme } from '../../../styles/theme';

export const CardStyled = styled(Link)`
    flex: 0 0 25%;
    padding: 12px;
    border: 1px solid ${theme.colors.borders.lightgrey};
    text-decoration: none;
    display: flex;
    flex-flow: column;
    height: 100%;
    background: ${theme.colors.main.white};
    position: relative;
    margin: 0 20px;
    min-height: 100px;

    &:hover {
        z-index: 2;
        background: ${theme.colors.main.background};
    }
`;

export const ImageStyled = styled.span<{ $image: string }>`
    display: block;
    background: url('${({ $image }) => $image}') no-repeat center / contain;
    width: 100%;
    flex: 1 1 30%;
    height: 100%;
`;
