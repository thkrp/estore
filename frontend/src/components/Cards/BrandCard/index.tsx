import React, { FC } from 'react';
import { Brand } from 'app-shared';
import { imageUrl } from '../../../common/helpers/path.helper';
import { CardStyled, ImageStyled } from './index.styles';

interface Props {
    item?: Brand;
    className?: string;
}

const BrandCard: FC<Props> = ({ item, className }) => {
    if (!item) {
        return null;
    }

    return (
        <CardStyled to={item.url} key={item.code} className={className} role="link">
            <ImageStyled $image={imageUrl(item.image)} />
        </CardStyled>
    );
};

export default BrandCard;
