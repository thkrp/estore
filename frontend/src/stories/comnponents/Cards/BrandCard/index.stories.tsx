import { Meta, StoryObj } from '@storybook/react';
import { mockData } from 'app-shared';
import { withRouter } from 'storybook-addon-react-router-v6';
import React from 'react';
import { BrandCard as BrandCardComponent } from '../../../../components';
import '../../../styles.css';

const meta = {
    title: 'Components/Cards/BrandCard',
    component: BrandCardComponent,
    decorators: [withRouter],
    parameters: {
        layout: 'wide'
    },
    tags: ['autodocs'],
    argTypes: {
        className: {
            description: 'class names'
        }
    }
} satisfies Meta<typeof BrandCardComponent>;

export default meta;
type Story = StoryObj<typeof meta>;
export const BrandCard: Story = {
    args: {
        item: mockData.brands[0]
    },
    decorators: [
        Story => (
            <div className="storybook-cards">
                <Story />
            </div>
        )
    ]
};
