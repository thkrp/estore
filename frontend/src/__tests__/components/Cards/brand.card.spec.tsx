/* eslint-disable no-restricted-imports*/
import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Brand, mockData } from 'app-shared';
import * as Sanitizer from '@braintree/sanitize-url';
import BrandCard from '../../../components/Cards/BrandCard';

describe('Brand card component', () => {
    let brand: Brand;
    beforeAll(() => {
        [brand] = mockData.brands;
    });

    it('should return null', () => {
        const { container } = render(<BrandCard />);
        expect(container).toBeEmptyDOMElement();
    });

    it('should return a card', () => {
        const { container } = render(<BrandCard item={brand} />, { wrapper: MemoryRouter });
        expect(container).toMatchSnapshot();
    });

    it('should have a sanitized link', () => {
        const sanitizeUrl = jest.spyOn(Sanitizer, 'sanitizeUrl');
        render(<BrandCard item={brand} />, { wrapper: MemoryRouter });
        expect(sanitizeUrl).toBeCalled();
    });

    it('should be able to pass a class', () => {
        render(<BrandCard item={brand} className="brand" />, { wrapper: MemoryRouter });
        const element = screen.getByRole('link');
        const classes = element.className.split(' ');
        expect(classes.includes('brand')).toBe(true);
    });
});
