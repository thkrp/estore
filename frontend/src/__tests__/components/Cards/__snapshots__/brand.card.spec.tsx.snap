// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brand card component should return a card 1`] = `
<div>
  <a
    class="sc-bcPKhP iTEbzn"
    href="about:blank"
    role="link"
  >
    <span
      class="sc-grXZZQ bgGREK"
    />
  </a>
</div>
`;
